ractor_cluster/src/node/client.rs

```diff
@@ -51,6 +54,76 @@ impl From<MessagingErr> for ClientConnectErr {
     }
 }
 
+/// TODO Either move this into ClientConnectErr or implement `std::error::Error::cause`
+#[derive(Debug)]
+enum NodeServerConnectionError {
+    Timeout(Duration),
+}
+
+impl Display for NodeServerConnectionError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            NodeServerConnectionError::Timeout(timeout) => {
+                write!(f, "Timeout: {}ms", timeout.as_millis())
+            }
+        }
+    }
+}
+
+impl std::error::Error for NodeServerConnectionError {}
+
+/// Connect to another [super::NodeServer] instance and wait for sessions to propagate.
+///
+/// * `node_server` - The [super::NodeServer] which will own this new connection session
+/// * `address` - The network address to send the connection to. Must implement [ToSocketAddrs]
+/// * `timeout` - The maximum duration to wait for sessions to propagate
+///
+/// Returns: [Ok(())] if the connection was successful and the [super::NodeSession] was started. Handshake will continue
+/// automatically. Results in a [Err(ClientConnectError)] if any part of the process failed to initiate
+pub async fn connect_and_verify<T>(
+    nodeserver: &ActorRef<super::NodeServer>,
+    addr: T,
+    timeout: Duration,
+) -> Result<(), super::ActorProcessingErr>
+    where
+        T: ToSocketAddrs,
+{
+    connect(nodeserver, &addr).await?;
+
+    sleep(Duration::from_millis(100)).await;
+
+    let tic = Instant::now();
+
+    loop {
+        let rpc_reply = call_t!(
```

Comment:
So you're not sleeping in here, which is a TON of requests to the actor.

Reply:
Oops! Good catch. Adjusted this so that the sleep is inside the loop